src/db.rs

```diff
@@ -0,0 +1,42 @@
+use postgres::{Client, Error, NoTls};
+
+pub struct Database {
+    user: String,
+    pass: String,
+    url: String,
+    port: String,
+}
+
+impl Database {
+    pub fn new(user: String, pass: String, url: String, port: String) -> Database {
+        Database {
+            user,
+            pass,
+            url,
+            port,
+        }
+    }
+    pub fn connect(&self) -> Result<(), Error> {
+        // Connect to postgresql client
+        let connection_link = format!(
+            "postgresql://{}:{}@{}:{}/pomegranate",
+            self.user, self.pass, self.url, self.port
+        );
```

Comment:
It might be cleaner to directly instantiate a `Config` struct. (See [here](https://docs.rs/postgres/0.19.5/postgres/config/struct.Config.html))

Reply:
Something like : connection_link = new Config(postgres_user, postgres_passwd, ...) ?

Reply:
There doesn't seem to be such a function implemented, but you can manually set the different properties using `user(...)`, `password(...)`...